### PR TITLE
update treelite to 2.1.0 and fix msvc build

### PR DESCRIPTION
**Pull request by Earthson**
treelite 2.0+ reduced dependencies: dmlc-core/opemmp.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -11,8 +11,14 @@ fn build_lib() {
         .build();
     println!("cargo:rustc-link-search={}/lib", dst.display());
     println!("cargo:rustc-link-lib=static=treelite_runtime_static");
-    println!("cargo:rustc-link-lib=stdc++");
-    println!("cargo:rustc-link-lib=gomp");
+    #[cfg(target_os = "linux")]
```
Comment (dovahcrow) [inline build.rs:14]: Does windows require a `cargo:rustc-link-lib` as well?

**Comment by dovahcrow** [comment]
This is great! I didn't expect the upgrading requires changing a single file only.

File: build.rs (hunk 1 above)
Comment (Earthson) [reply build.rs:14]: toolchain=msvc: Windows requires MSVC runtime. It works fine without a configuration like "cargo:rustc-link-lib=stdc++".

Comment (dovahcrow) [reply build.rs:14]: That's cool!

**Comment by dovahcrow** [comment]
Thanks for your help! Much appreciated!

**Pull request merged**